### DISCUSSION TITLE
Directory Enforcement

### DISCUSSION THREAD

**Discussion by Larry-Home** [awcodes/filament-curator#200] [Ideas]
So, we have a number of "sections" on our web site that are maintained by different groups.  Inside the CuratorPicker component we are able to specify a directory where images are stored and this is working as expected.  However, it is not filtering images that are only in that directory and there does not seem to be an option to restrict it to only those images in the Media Picker.

Likewise, it would be nice if we could setup multiple Media navigation items that would allow for management of the images only in a specified folder.  If no folder is specified it would be nice if the media was organized in this file structure so that you could easily see where images are stored.

The component was very easy to get going and works very well, so thanks for that!  Coupled with the fact it is actively being developed is great.

**Comment by awcodes** [comment]
Interesting. So you need the ability to restrict media to a multi-context application? That could get really complicated really quickly. Should be possible out of the box with a multi db solution, but I can see where it breaks down with a single db.

**Comment by awcodes** [comment]
I'm open to ideas and PRs is you have a good solution.

**Comment by Larry-Home** [comment]
It is simpler like this in my case.  We have an Events page and all the photos for that are stored in one folder.  Then we have an opportunities page and the images for that are stored in another folder.  When a person that is in charge of events goes to grab or add a picture it would be nice if one the events ones were displayed.

**Comment by awcodes** [comment]
Ok. Yea, I'm not sure of a clean way to enforce that, but let me try some things and see if I can come up with something.

**Comment by awcodes** [comment]
OK. v2.7.0 should get you what you need. You can now use `->limitToDirectory()` on the Picker field to restrict the items in the modal to the directory assigned to that Picker field.

**Reply by Larry-Home** [reply]
WOW, that was fast!  

Here are my results with testing it based on the signature in the PR: 

- If I just add it as shown or passing a value of true I end up getting: No Files in the library or nothing found for your search.  
- If I pass it with null / false then I get all of the media items being displayed.
- If I pass it the same string that I would pass into directory then it works as needed.

Thank you so much for making this available!

**Reply by awcodes** [reply]
You shouldn't need to pass it anything it defaults to true and uses what ever is set with ->directory(). If you don't supply a directory it just shows everything as normal.

**Reply by awcodes** [reply]
So you'd just set ->directory() on each media picker you need to limit. And then you have the option to either limit it at picker level or globally to make all pickers limit.

**Reply by Larry-Home** [reply]
Actually that is working though I swear it gave me No Files...

**Comment by Larry-Home** [comment]
Again, thanks for adding this and providing this excellent plugin!

**Reply by awcodes** [reply]
Thank you for using it and enjoying it.
